File: lcm_optimised.cpp
#include <iostream>
#include<algorithm>
using namespace std;


int gcd(int a, int b)
{
    if(a==0 || b==0)
        return max(a,b);
    else
    {
        if(max(a,b)%min(a,b) == 0)
            return min(a,b);
        else
            return gcd(min(a,b), max(a,b)%min(a,b));
            
    }
}


int  lcm_fast(int a, int b) {
   int lcm = (a*b)/gcd(a,b);
   return lcm;
}







int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout <<lcm_fast(a, b) << std::endl;
  return 0;
}

File: lcm_optimised.py
# Uses python3
import sys

def gcd(a, b):
    if(a==0 or b==0):
        return max(a,b)
    else:
        if(max(a,b)%min(a,b) == 0):
            return min(a,b)
        else:
            return(gcd(min(a,b),max(a,b)%min(a,b)))




def lcm_naive(a,b):
    lcm = int((a*b)/gcd(a,b))
    return lcm

if __name__ == '__main__':
    input = sys.stdin.read()
    a, b = map(int, input.split())
    #print(gcd(a,b))
    print(lcm_naive(a, b))

